mlflow/utils/uri.py

```diff
@@ -162,12 +170,57 @@ def extract_db_type_from_uri(db_uri):
     return db_type
 
 
+def _is_local_windows_path(uri_or_path):
+    """
+    Detects if platform system is windows and if `uri_or_path` is a windows path
+    """
+    parsed_uri = urllib.parse.urlparse(uri_or_path)
+    scheme = parsed_uri.scheme
+    return (
+        platform.system().lower() == "windows"
+        and not parsed_uri.netloc
+        and uri_or_path[0] in string.ascii_letters
+        and uri_or_path[0].lower() == scheme.lower()
+        and uri_or_path[1:3] in [":", ":/", "://", ":\\"]
+    )
+
+
+def _strip_leading_slash_if_windows_path(local_path):
+    """
+    Removes leading slash, if platform system is windows and `local_path` is a windows path
+
+    # For Windows path, urllib parses it with an extra leading slash
```

Comment:
Is this behavior specific to Windows?

```python
>>> urllib.parse.urlparse("file:////a/b/c")
ParseResult(scheme='file', netloc='', path='//a/b/c', params='', query='', fragment='')
>>> urllib.parse.urlparse("file:////C:/a/b/c")
ParseResult(scheme='file', netloc='', path='//C:/a/b/c', params='', query='', fragment='')
```

Reply:
yes, only specific to windows.

Reply:
...because POSIX fs absolute path will always have a leading slash anyways, but on windows, a drive letter is needed and not a leading slash.

on posix system:
```
import pathlib
print(pathlib.Path("/a/b/c").is_absolute())
```
ouput:
`True`